src/main/java/org/folio/rest/helper/TransactionsHelper.java

```diff
@@ -90,4 +97,34 @@ private void validateTransactionType(Transaction transaction, Transaction.Transa
     return updateTransaction(transaction);
   }
 
+  private CompletableFuture<Transaction> checkRestrictions(Transaction transaction) {
+    CompletableFuture<Transaction> future = new VertxCompletableFuture<>(ctx);
+    switch(transaction.getTransactionType()) {
+      case ALLOCATION:
+      case TRANSFER:
+        if ((Objects.isNull(transaction.getFromFundId()) ^ Objects.isNull(transaction.getToFundId())) &&
+          transaction.getTransactionType().equals(ALLOCATION)) {
+          future.complete(transaction);
+        } else if (Objects.nonNull(transaction.getFromFundId()) && Objects.nonNull(transaction.getToFundId())) {
+          checkAllocatedIds(transaction)
+            .thenApply(isMatch ->
+              Boolean.TRUE.equals(isMatch) ? future.complete(transaction) : future.completeExceptionally(new HttpException(422, ALLOCATION_IDS_MISMATCH)))
+            .exceptionally(throwable -> future.completeExceptionally(new HttpException(500, ALLOCATION_TRANSFER_FAILED)));
```

Comment:
ALLOCATION_TRANSFER_FAILED error code is used for Allocation and Transfer, but description says "Failed to create allocation transaction for a budget"

Reply:
removed